src/Microsoft.Identity.Web/DownstreamWebApiSupport/DownstreamWebApi.cs

```diff
@@ -13,208 +13,208 @@
-        {
-            // Gets the options from configuration (or default value)
-            DownstreamWebApiOptions options;
-            if (optionsInstanceName != null)
-            {
-                options = _namedDownstreamWebApiOptions.Get(optionsInstanceName);
-            }
-            else
-            {
-                options = _namedDownstreamWebApiOptions.CurrentValue;
-            }
-
-            DownstreamWebApiOptions clonedOptions = options.Clone();
-            calledApiOptionsOverride?.Invoke(clonedOptions);
-            return clonedOptions;
-        }
-
-        private static void CreateProofOfPossessionConfiguration(DownstreamWebApiOptions effectiveOptions, string apiUrl)
-        {
-            if (effectiveOptions.IsProofOfPossessionRequest && effectiveOptions.TokenAcquisitionOptions?.PoPConfiguration != null)
-            {
-                if (effectiveOptions.TokenAcquisitionOptions == null)
-                {
-                    effectiveOptions.TokenAcquisitionOptions = new TokenAcquisitionOptions();
-                }
-
-                effectiveOptions.TokenAcquisitionOptions.PoPConfiguration = new PoPAuthenticationConfiguration(new Uri(apiUrl))
-                {
-                    HttpMethod = effectiveOptions.HttpMethod,
-                };
-            }
-        }
-    }
+		}
+
+		/// <inheritdoc/>
+		public async Task<HttpResponseMessage> CallWebApiForUserAsync(
+			string serviceName,
+			Action<DownstreamWebApiOptions>? calledDownstreamWebApiOptionsOverride = null,
+			ClaimsPrincipal? user = null,
+			StringContent? content = null)
+		{
+			DownstreamWebApiOptions effectiveOptions = MergeOptions(serviceName, calledDownstreamWebApiOptionsOverride);
+
+			if (string.IsNullOrEmpty(effectiveOptions.Scopes))
+			{
+				throw new ArgumentException(IDWebErrorMessage.ScopesNotConfiguredInConfigurationOrViaDelegate);
+			}
+
+			string apiUrl = effectiveOptions.GetApiUrl();
+
+			CreateProofOfPossessionConfiguration(effectiveOptions, apiUrl);
+
+			string? userflow;
+			if (_microsoftIdentityOptions.IsB2C && string.IsNullOrEmpty(effectiveOptions.UserFlow))
+			{
+				userflow = _microsoftIdentityOptions.DefaultUserFlow;
+			}
+			else
+			{
+				userflow = effectiveOptions.UserFlow;
+			}
+
+			AuthenticationResult authResult = await _tokenAcquisition.GetAuthenticationResultForUserAsync(
+				effectiveOptions.GetScopes(),
+				effectiveOptions.Tenant,
+				userflow,
+				user,
+				effectiveOptions.TokenAcquisitionOptions)
+				.ConfigureAwait(false);
+
+			HttpResponseMessage response;
+			using (HttpRequestMessage httpRequestMessage = new HttpRequestMessage(
+				effectiveOptions.HttpMethod,
+				apiUrl))
+			{
+				if (content != null)
+				{
+					httpRequestMessage.Content = content;
+				}
+
+				httpRequestMessage.Headers.Add(
+					Constants.Authorization,
+					authResult.CreateAuthorizationHeader());
+				response = await _httpClient.SendAsync(httpRequestMessage).ConfigureAwait(false);
```

Comment:
the response var isn't used outside the scope of the using block. It can be removed and replaced by a direct return statement in front of the awaited call.